src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Services/InMemoryCacheServiceTests.cs

```diff
@@ -0,0 +1,441 @@
+
+        memoryCache
+            .Setup(mock => mock.TryGetValue(cacheKey, out entity))
+            .Returns(true);
+
+        var service = SetupService(memoryCache.Object);
+
+        var result = await service.GetItem(cacheKey, typeof(SampleClass));
+        VerifyAllMocks(memoryCache);
+
+        Assert.Equal(entity, result);
+    }
+
+    [Fact]
+    public async Task GetItem_NullIfLessSpecificSupertype()
+    {
+        object entity = new SampleClassSuperclass();
+
+        var cacheKey = new SampleCacheKey("Key");
+
+        var memoryCache = new Mock<IMemoryCache>(Strict);
+
+        memoryCache
+            .Setup(mock => mock.TryGetValue(cacheKey, out entity))
+            .Returns(true);
+
+        var service = SetupService(memoryCache.Object);
+
+        var result = await service.GetItem(cacheKey, typeof(SampleClass));
+        VerifyAllMocks(memoryCache);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task GetItem_NullIfCacheMiss()
+    {
+        var cacheKey = new SampleCacheKey("Key");
+
+        var memoryCache = new Mock<IMemoryCache>(Strict);
+
+        object entity;
+            
+        memoryCache
+            .Setup(mock => mock.TryGetValue(cacheKey, out entity))
+            .Returns(false);
+
+        var service = SetupService(memoryCache.Object);
+
+        var result = await service.GetItem(cacheKey, typeof(SampleClass));
+        VerifyAllMocks(memoryCache);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task GetItem_NullIfException()
+    {
+        var cacheKey = new SampleCacheKey("");
+
+        var memoryCache = new Mock<IMemoryCache>(Strict);
+
+        object entity;
+        memoryCache
+            .Setup(mock => mock.TryGetValue(cacheKey, out entity))
+            .Throws(new Exception("Something went wrong"));
+
+        var service = SetupService(memoryCache.Object);
+
+        var result = await service.GetItem(cacheKey, typeof(SampleClass));
+        VerifyAllMocks(memoryCache);
+
+        Assert.Null(result);
+    }
+
+    [Fact]
+    public async Task GetItem_NullIfIncorrectType()
+    {
+        object entityOfIncorrectType = new SampleClass();
+
+        var cacheKey = new SampleCacheKey("");
+
+        var memoryCache = new Mock<IMemoryCache>(Strict);
+
+        memoryCache
+            .Setup(mock => mock.TryGetValue(cacheKey, out entityOfIncorrectType))
+            .Returns(true);
+
+        var service = SetupService(memoryCache.Object);
+
+        var result = await service.GetItem(cacheKey, typeof(string));
+        VerifyAllMocks(memoryCache);
+
+        Assert.Null(result);
+    }
+
+    private static InMemoryCacheService SetupService(
+        IMemoryCache? memoryCache = null,
+        ILogger<InMemoryCacheService>? logger = null)
+    {
+        var service = new InMemoryCacheService(
+            logger ?? Mock.Of<ILogger<InMemoryCacheService>>()
+        );
+        service.SetMemoryCache(memoryCache ?? Mock.Of<IMemoryCache>());
```

Comment:
I've made it a simple constructor arg now, the default of which is created manually in `Startup.cs` of the Content API project.

Additionally, I've added some appsettings config to control the cache behaviour settings in case we want to test and tweak these settings without multiple redeploys (and additionally to be able to disable it if necessary).